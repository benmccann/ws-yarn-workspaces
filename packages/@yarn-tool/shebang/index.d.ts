export declare const reShebang: RegExp;
/**
 * Extract normalized shebang command token.
 *
 *
 * Examples:
 *
 *  shebang("#!/usr/bin/ruby") // "ruby"
 *
 *  shebang("#!/usr/bin/env node") // "node"
 *
 *  @param: {String}
 *  @return {String|null}
 */
export declare function shebang(line: string): IScriptName;
export declare function removeShebang(line: string): string;
export declare type IScriptName = string | 'env' | 'node' | 'sh';
export interface IMatchShebangReturnType {
    shebang: string;
    bin: string;
    name: IScriptName;
    script: IScriptName;
    argv: string | '';
    isExtra: boolean;
}
export declare function matchShebang(line: string): IMatchShebangReturnType;
export default shebang;
