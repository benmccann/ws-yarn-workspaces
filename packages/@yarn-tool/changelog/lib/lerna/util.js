"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const lodash_1 = require("lodash");
function handleOptions(options) {
    options = lodash_1.defaults(options !== null && options !== void 0 ? options : {}, {
        type: 'independent',
        changelogPreset: '@bluelovers/conventional-changelog-bluelovers',
        tagPrefix: 'v',
    });
    if (!options.changelogPreset) {
        options.changelogPreset = tryRequire('@bluelovers/conventional-changelog-bluelovers') ? '@bluelovers/conventional-changelog-bluelovers' : 'conventional-changelog-angular';
    }
    return options;
}
exports.handleOptions = handleOptions;
function tryRequire(name) {
    try {
        return require.resolve(name);
    }
    catch (err) {
        if (err.code !== "MODULE_NOT_FOUND") {
            throw new err;
        }
    }
}
//# sourceMappingURL=util.js.map