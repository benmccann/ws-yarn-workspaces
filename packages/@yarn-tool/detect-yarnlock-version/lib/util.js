"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMetadataVersion = getMetadataVersion;
exports._getMetadataVersionCore = _getMetadataVersionCore;
function getMetadataVersion(yarnLockObject) {
    return _getMetadataVersionCore(yarnLockObject.__metadata);
}
function _getMetadataVersionCore(__metadata) {
    var _a;
    return (_a = __metadata === null || __metadata === void 0 ? void 0 : __metadata.version) === null || _a === void 0 ? void 0 : _a.toString();
}
//# sourceMappingURL=util.js.map