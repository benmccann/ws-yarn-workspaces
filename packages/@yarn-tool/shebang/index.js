"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reShebang = void 0;
exports.shebang = shebang;
exports.removeShebang = removeShebang;
exports.matchShebang = matchShebang;
exports.reShebang = /^\s*#!\s*([^\s]+)(?:\s+([^\s]+)(?:\s+([^\r\n]+))?(?=\r|\n|$))?/;
/**
 * Extract normalized shebang command token.
 *
 *
 * Examples:
 *
 *  shebang("#!/usr/bin/ruby") // "ruby"
 *
 *  shebang("#!/usr/bin/env node") // "node"
 *
 *  @param: {String}
 *  @return {String|null}
 */
function shebang(line) {
    var _a;
    return (_a = matchShebang(line)) === null || _a === void 0 ? void 0 : _a.script;
}
function removeShebang(line) {
    return line.replace(exports.reShebang, '');
}
function matchShebang(line) {
    const matched = line.match(exports.reShebang);
    if ((matched === null || matched === void 0 ? void 0 : matched.length) > 0) {
        const paths = matched[1].split('/');
        let script = paths[paths.length - 1];
        let name = script;
        let i = 2;
        let isExtra = false;
        if (script === 'env' && typeof matched[2] !== 'undefined') {
            script = matched[2];
            i = 3;
            isExtra = true;
        }
        let rest = matched.slice(i).filter(v => typeof v !== 'undefined');
        return {
            shebang: matched[0],
            bin: matched[1],
            name,
            script,
            argv: rest.join(' '),
            isExtra,
        };
    }
}
exports.default = shebang;
//# sourceMappingURL=index.js.map