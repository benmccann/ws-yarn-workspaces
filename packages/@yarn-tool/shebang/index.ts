export const reShebang = /^\s*#!\s*([^\s]+)(?:\s+([^\s]+)(?:\s+([^\r\n]+))?(?=\r|\n|$))?/;

/**
 * Extract normalized shebang command token.
 *
 *
 * Examples:
 *
 *  shebang("#!/usr/bin/ruby") // "ruby"
 *
 *  shebang("#!/usr/bin/env node") // "node"
 *
 *  @param: {String}
 *  @return {String|null}
 */
export function shebang(line: string): IScriptName
{
	return matchShebang(line)?.script
}

export function removeShebang(line: string)
{
	return line.replace(reShebang, '')
}

export type IScriptName = string | 'env' | 'node' | 'sh';

export interface IMatchShebangReturnType
{
	shebang: string;
	bin: string;
	name: IScriptName;
	script: IScriptName;
	argv: string | '';
	isExtra: boolean;
}

export function matchShebang(line: string): IMatchShebangReturnType
{
	const matched = line.match(reShebang)

	if (matched?.length > 0)
	{
		const paths = matched[1].split('/');
		let script: IScriptName = paths[paths.length - 1];
		let name: IScriptName = script;

		let i = 2;
		let isExtra: boolean = false;

		if (script === 'env')
		{
			script = matched[2];
			i = 3;
			isExtra = true;
		}

		let rest = matched.slice(i).filter(v => typeof v !== 'undefined')

		return {
			shebang: matched[0],

			bin: matched[1],
			name,

			script,
			argv: rest.join(' '),

			isExtra,
		}
	}
}

export default shebang;
