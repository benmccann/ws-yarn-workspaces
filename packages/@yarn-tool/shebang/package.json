{
  "name": "@yarn-tool/shebang",
  "version": "1.0.14",
  "description": "Extract normalized shebang command token",
  "keywords": [
    "shebang",
    "lang",
    "language",
    "command",
    "regex",
    "regexp",
    "match",
    "test",
    "line",
    "cli",
    "bin",
    "strip",
    "remove",
    "hasbang",
    "hash",
    "pound"
  ],
  "homepage": "https://github.com/bluelovers/ws-yarn-workspaces/tree/master/packages/@yarn-tool/shebang#readme",
  "bugs": {
    "url": "https://github.com/bluelovers/ws-yarn-workspaces/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bluelovers/ws-yarn-workspaces.git",
    "directory": "packages/@yarn-tool/shebang"
  },
  "license": "MIT",
  "author": "Alexandru Vladutu <alexandru.vladutu@gmail.com>",
  "main": "index.js",
  "directories": {
    "test": "test"
  },
  "scripts": {
    "test": "jest",
    "test:jest": "jest --passWithNoTests",
    "test:jest:snapshot": "yarn run test:jest -- -u",
    "test:snapshot": "yarn run test -- -u",
    "test:tsd": "ynpx tsd",
    "npm:publish": "npm publish",
    "npm:publish:lerna": "npx lerna publish --yes --bump patch",
    "preversion": "yarn run test",
    "prepublish:lockfile": "npx sync-lockfile .",
    "prepublishOnly": "echo prepublishOnly",
    "postpublish_": "git commit -m \"chore(release): publish & git push & echo postpublish\" .",
    "ncu": "npx yarn-tool ncu -u",
    "sort-package-json": "npx yarn-tool sort"
  },
  "dependencies": {
    "tslib": ">=2"
  },
  "publishConfig": {
    "access": "public"
  },
  "gitHead": "e077fb60fdf2c7106ac66bd9fcc0f734c5db5313"
}