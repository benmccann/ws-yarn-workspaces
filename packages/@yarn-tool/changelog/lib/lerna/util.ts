import {
	IOptionsWithType,
	IOptionsUpdateChangelog,
	IOptionsRecommendVersion,
	IOptionsBaseCore,
	IChangelogPreset, IOptionsBase,
} from '../types';
import { defaults } from 'lodash';
import { requireResolveExtra, requireResolveCore } from '@yarn-tool/require-resolve';
import { join } from 'path';

export function handleOptions<T extends IOptionsRecommendVersion | IOptionsUpdateChangelog>(options?: IOptionsWithType<T>): IOptionsWithType<T>
{
	const defaultChangelogPreset = '@bluelovers/conventional-changelog-bluelovers' as const;

	options = defaults(options ?? {} as null, <IOptionsWithType<IOptionsBase>>{
		type: 'independent',
		changelogPreset: void 0,
		tagPrefix: 'v',
	})

	if (typeof options.type !== 'string' || !options.type.length)
	{
		options.type = 'independent'
	}

	if (typeof options.tagPrefix !== 'string')
	{
		options.tagPrefix = 'v'
	}

	if (typeof options.changelogPreset !== 'string' || !options.changelogPreset.length || !Boolean(options.changelogPreset))
	{
		options.changelogPreset = void 0;
	}

	if (!options.changelogPreset || options.changelogPreset === defaultChangelogPreset)
	{
		options.changelogPreset = requireResolveExtra(defaultChangelogPreset, {
			includeGlobal: true,
			includeCurrentDirectory: true,
			paths: [
				join(__dirname, '../..'),
			],
		}).result ?? options.changelogPreset;
	}

	options.changelogPreset ??= 'conventional-changelog-angular';

	return options
}

function tryRequire(name: IChangelogPreset)
{
	try
	{
		return requireResolveCore(name, {
			includeGlobal: true,
			includeCurrentDirectory: true,
		})
	}
	catch (err)
	{
		if (err.code !== "MODULE_NOT_FOUND")
		{
			throw new err
		}
	}
}
