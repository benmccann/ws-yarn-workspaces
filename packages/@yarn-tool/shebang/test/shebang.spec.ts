import shebang, { matchShebang, removeShebang, reShebang } from '../';
import { inspect } from 'util';

const data = {
	"/usr/bin/ruby": "ruby",
	"/usr/bin/env node": "node",
	"/usr/bin/env python -c": "python",
	"/bin/sh": "sh",
	"/usr/awk -f": "awk",
	"/bin/sed -f": "sed",
	"/usr/bin/php": "php",
	"/usr/bin/php5": "php5",
};

describe(`shebang`, () =>
{
	for (const _prop in data)
	{
		const expected = data[_prop];

		[
			`#!${_prop}`,
			`#! ${_prop}`,

			`\n\n#!${_prop}\n\n`,
			`\n\n#! ${_prop}\n\n`,

			`\n\n#!${_prop}`,
			`\n\n#! ${_prop}`,

			`#!${_prop}\n\n`,
			`#! ${_prop}\n\n`,
		].forEach((prop, index) => {

			test(`${inspect(prop)} => ${expected}`, () =>
			{
				let actual: unknown;

				actual = shebang(prop);

				expect(actual).toStrictEqual(expected);

				actual = matchShebang(prop);

				expect(actual).toHaveProperty('bin', _prop.split(' ')[0])

				expect(actual).toMatchSnapshot();

				expect(removeShebang(prop)).not.toMatch(reShebang);

				expect(removeShebang(prop)).toMatchSnapshot();
			})

		})
		;

	}
});

describe(`should not found`, () =>
{
	for (const prop in data)
	{
		test(`${inspect(prop)}`, () =>
		{
			let actual: unknown;

			actual = shebang(prop);

			expect(actual).toBeUndefined();

			actual = matchShebang(prop);

			expect(actual).toBeUndefined();
		})
	}
})
