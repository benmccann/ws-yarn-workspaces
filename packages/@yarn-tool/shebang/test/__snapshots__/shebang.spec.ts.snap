// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shebang '#! /bin/sed -f' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "#! /bin/sed -f",
}
`;

exports[`shebang '#! /bin/sed -f' => sed 2`] = `""`;

exports[`shebang '#! /bin/sed -f\\n\\n' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "#! /bin/sed -f",
}
`;

exports[`shebang '#! /bin/sed -f\\n\\n' => sed 2`] = `
"

"
`;

exports[`shebang '#! /bin/sh' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "#! /bin/sh",
}
`;

exports[`shebang '#! /bin/sh' => sh 2`] = `""`;

exports[`shebang '#! /bin/sh\\n\\n' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "#! /bin/sh",
}
`;

exports[`shebang '#! /bin/sh\\n\\n' => sh 2`] = `
"

"
`;

exports[`shebang '#! /usr/awk -f' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "#! /usr/awk -f",
}
`;

exports[`shebang '#! /usr/awk -f' => awk 2`] = `""`;

exports[`shebang '#! /usr/awk -f\\n\\n' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "#! /usr/awk -f",
}
`;

exports[`shebang '#! /usr/awk -f\\n\\n' => awk 2`] = `
"

"
`;

exports[`shebang '#! /usr/bin/env node' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "#! /usr/bin/env node",
}
`;

exports[`shebang '#! /usr/bin/env node' => node 2`] = `""`;

exports[`shebang '#! /usr/bin/env node\\n\\n' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "#! /usr/bin/env node",
}
`;

exports[`shebang '#! /usr/bin/env node\\n\\n' => node 2`] = `
"

"
`;

exports[`shebang '#! /usr/bin/env python -c' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "#! /usr/bin/env python -c",
}
`;

exports[`shebang '#! /usr/bin/env python -c' => python 2`] = `""`;

exports[`shebang '#! /usr/bin/env python -c\\n\\n' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "#! /usr/bin/env python -c",
}
`;

exports[`shebang '#! /usr/bin/env python -c\\n\\n' => python 2`] = `
"

"
`;

exports[`shebang '#! /usr/bin/php' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "#! /usr/bin/php",
}
`;

exports[`shebang '#! /usr/bin/php' => php 2`] = `""`;

exports[`shebang '#! /usr/bin/php\\n\\n' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "#! /usr/bin/php",
}
`;

exports[`shebang '#! /usr/bin/php\\n\\n' => php 2`] = `
"

"
`;

exports[`shebang '#! /usr/bin/php5' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "#! /usr/bin/php5",
}
`;

exports[`shebang '#! /usr/bin/php5' => php5 2`] = `""`;

exports[`shebang '#! /usr/bin/php5\\n\\n' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "#! /usr/bin/php5",
}
`;

exports[`shebang '#! /usr/bin/php5\\n\\n' => php5 2`] = `
"

"
`;

exports[`shebang '#! /usr/bin/ruby' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "#! /usr/bin/ruby",
}
`;

exports[`shebang '#! /usr/bin/ruby' => ruby 2`] = `""`;

exports[`shebang '#! /usr/bin/ruby\\n\\n' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "#! /usr/bin/ruby",
}
`;

exports[`shebang '#! /usr/bin/ruby\\n\\n' => ruby 2`] = `
"

"
`;

exports[`shebang '#!/bin/sed -f' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "#!/bin/sed -f",
}
`;

exports[`shebang '#!/bin/sed -f' => sed 2`] = `""`;

exports[`shebang '#!/bin/sed -f\\n\\n' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "#!/bin/sed -f",
}
`;

exports[`shebang '#!/bin/sed -f\\n\\n' => sed 2`] = `
"

"
`;

exports[`shebang '#!/bin/sh' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "#!/bin/sh",
}
`;

exports[`shebang '#!/bin/sh' => sh 2`] = `""`;

exports[`shebang '#!/bin/sh\\n\\n' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "#!/bin/sh",
}
`;

exports[`shebang '#!/bin/sh\\n\\n' => sh 2`] = `
"

"
`;

exports[`shebang '#!/usr/awk -f' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "#!/usr/awk -f",
}
`;

exports[`shebang '#!/usr/awk -f' => awk 2`] = `""`;

exports[`shebang '#!/usr/awk -f\\n\\n' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "#!/usr/awk -f",
}
`;

exports[`shebang '#!/usr/awk -f\\n\\n' => awk 2`] = `
"

"
`;

exports[`shebang '#!/usr/bin/env node' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "#!/usr/bin/env node",
}
`;

exports[`shebang '#!/usr/bin/env node' => node 2`] = `""`;

exports[`shebang '#!/usr/bin/env node\\n\\n' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "#!/usr/bin/env node",
}
`;

exports[`shebang '#!/usr/bin/env node\\n\\n' => node 2`] = `
"

"
`;

exports[`shebang '#!/usr/bin/env python -c' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "#!/usr/bin/env python -c",
}
`;

exports[`shebang '#!/usr/bin/env python -c' => python 2`] = `""`;

exports[`shebang '#!/usr/bin/env python -c\\n\\n' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "#!/usr/bin/env python -c",
}
`;

exports[`shebang '#!/usr/bin/env python -c\\n\\n' => python 2`] = `
"

"
`;

exports[`shebang '#!/usr/bin/php' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "#!/usr/bin/php",
}
`;

exports[`shebang '#!/usr/bin/php' => php 2`] = `""`;

exports[`shebang '#!/usr/bin/php\\n\\n' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "#!/usr/bin/php",
}
`;

exports[`shebang '#!/usr/bin/php\\n\\n' => php 2`] = `
"

"
`;

exports[`shebang '#!/usr/bin/php5' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "#!/usr/bin/php5",
}
`;

exports[`shebang '#!/usr/bin/php5' => php5 2`] = `""`;

exports[`shebang '#!/usr/bin/php5\\n\\n' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "#!/usr/bin/php5",
}
`;

exports[`shebang '#!/usr/bin/php5\\n\\n' => php5 2`] = `
"

"
`;

exports[`shebang '#!/usr/bin/ruby' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "#!/usr/bin/ruby",
}
`;

exports[`shebang '#!/usr/bin/ruby' => ruby 2`] = `""`;

exports[`shebang '#!/usr/bin/ruby\\n\\n' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "#!/usr/bin/ruby",
}
`;

exports[`shebang '#!/usr/bin/ruby\\n\\n' => ruby 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /bin/sed -f' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "

#! /bin/sed -f",
}
`;

exports[`shebang '\\n\\n#! /bin/sed -f' => sed 2`] = `""`;

exports[`shebang '\\n\\n#! /bin/sed -f\\n\\n' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "

#! /bin/sed -f",
}
`;

exports[`shebang '\\n\\n#! /bin/sed -f\\n\\n' => sed 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /bin/sh' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "

#! /bin/sh",
}
`;

exports[`shebang '\\n\\n#! /bin/sh' => sh 2`] = `""`;

exports[`shebang '\\n\\n#! /bin/sh\\n\\n' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "

#! /bin/sh",
}
`;

exports[`shebang '\\n\\n#! /bin/sh\\n\\n' => sh 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/awk -f' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "

#! /usr/awk -f",
}
`;

exports[`shebang '\\n\\n#! /usr/awk -f' => awk 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/awk -f\\n\\n' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "

#! /usr/awk -f",
}
`;

exports[`shebang '\\n\\n#! /usr/awk -f\\n\\n' => awk 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/bin/env node' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "

#! /usr/bin/env node",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/env node' => node 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/bin/env node\\n\\n' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "

#! /usr/bin/env node",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/env node\\n\\n' => node 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/bin/env python -c' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "

#! /usr/bin/env python -c",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/env python -c' => python 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/bin/env python -c\\n\\n' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "

#! /usr/bin/env python -c",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/env python -c\\n\\n' => python 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/bin/php' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "

#! /usr/bin/php",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/php' => php 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/bin/php\\n\\n' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "

#! /usr/bin/php",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/php\\n\\n' => php 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/bin/php5' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "

#! /usr/bin/php5",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/php5' => php5 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/bin/php5\\n\\n' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "

#! /usr/bin/php5",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/php5\\n\\n' => php5 2`] = `
"

"
`;

exports[`shebang '\\n\\n#! /usr/bin/ruby' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "

#! /usr/bin/ruby",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/ruby' => ruby 2`] = `""`;

exports[`shebang '\\n\\n#! /usr/bin/ruby\\n\\n' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "

#! /usr/bin/ruby",
}
`;

exports[`shebang '\\n\\n#! /usr/bin/ruby\\n\\n' => ruby 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/bin/sed -f' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "

#!/bin/sed -f",
}
`;

exports[`shebang '\\n\\n#!/bin/sed -f' => sed 2`] = `""`;

exports[`shebang '\\n\\n#!/bin/sed -f\\n\\n' => sed 1`] = `
Object {
  "argv": "-f",
  "bin": "/bin/sed",
  "script": "sed",
  "shebang": "

#!/bin/sed -f",
}
`;

exports[`shebang '\\n\\n#!/bin/sed -f\\n\\n' => sed 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/bin/sh' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "

#!/bin/sh",
}
`;

exports[`shebang '\\n\\n#!/bin/sh' => sh 2`] = `""`;

exports[`shebang '\\n\\n#!/bin/sh\\n\\n' => sh 1`] = `
Object {
  "argv": "",
  "bin": "/bin/sh",
  "script": "sh",
  "shebang": "

#!/bin/sh",
}
`;

exports[`shebang '\\n\\n#!/bin/sh\\n\\n' => sh 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/awk -f' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "

#!/usr/awk -f",
}
`;

exports[`shebang '\\n\\n#!/usr/awk -f' => awk 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/awk -f\\n\\n' => awk 1`] = `
Object {
  "argv": "-f",
  "bin": "/usr/awk",
  "script": "awk",
  "shebang": "

#!/usr/awk -f",
}
`;

exports[`shebang '\\n\\n#!/usr/awk -f\\n\\n' => awk 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/bin/env node' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "

#!/usr/bin/env node",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/env node' => node 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/bin/env node\\n\\n' => node 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/env",
  "script": "node",
  "shebang": "

#!/usr/bin/env node",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/env node\\n\\n' => node 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/bin/env python -c' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "

#!/usr/bin/env python -c",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/env python -c' => python 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/bin/env python -c\\n\\n' => python 1`] = `
Object {
  "argv": "-c",
  "bin": "/usr/bin/env",
  "script": "python",
  "shebang": "

#!/usr/bin/env python -c",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/env python -c\\n\\n' => python 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/bin/php' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "

#!/usr/bin/php",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/php' => php 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/bin/php\\n\\n' => php 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php",
  "script": "php",
  "shebang": "

#!/usr/bin/php",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/php\\n\\n' => php 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/bin/php5' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "

#!/usr/bin/php5",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/php5' => php5 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/bin/php5\\n\\n' => php5 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/php5",
  "script": "php5",
  "shebang": "

#!/usr/bin/php5",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/php5\\n\\n' => php5 2`] = `
"

"
`;

exports[`shebang '\\n\\n#!/usr/bin/ruby' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "

#!/usr/bin/ruby",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/ruby' => ruby 2`] = `""`;

exports[`shebang '\\n\\n#!/usr/bin/ruby\\n\\n' => ruby 1`] = `
Object {
  "argv": "",
  "bin": "/usr/bin/ruby",
  "script": "ruby",
  "shebang": "

#!/usr/bin/ruby",
}
`;

exports[`shebang '\\n\\n#!/usr/bin/ruby\\n\\n' => ruby 2`] = `
"

"
`;
